From 1a1ed7e7f6a312a11156722eb982feea9f4d984e Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Mon, 3 Sep 2018 01:21:07 +0100
Subject: [PATCH 19/19] soong: Add support for target specific headers

 * While in certain cases TARGET_SPECIFIC_HEADER_PATH still works,
   in many projects Android.mk has been converted to Android.bp
   and in those cases the headers provided by devices are no longer
   taken into account. Take the case of libcamera_client for
   example, given that a few devices provide a custom camera
   parameters lib. In that case, when the lib is statically linked
   the used headers are those from fw/av causing wreckage.

Change-Id: Ib78df67a944cec9a76a2d3cd08ea18276136b5d7
---
 build/soong/android/variable.go | 1 +
 build/soong/soong_config.mk     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index a6da3dd5..91234c52 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -60,6 +60,7 @@ type ProductVariables struct {
 	Target_exfat_driver  *string `json:",omitempty"`
 	Target_process_sdk_version_override *string `json:",omitempty"`
 	Target_shim_libs  *string `json:",omitempty"`
+	Target_specific_headers_include_dir  *string `json:",omitempty"`
 	Uses_generic_camera_parameter_library  *bool `json:",omitempty"`
 	Uses_nvidia_enhancements  *bool `json:",omitempty"`
 	Uses_qcom_bsp_legacy  *bool `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 0f3e9a0d..b07bbab0 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -14,6 +14,7 @@ $(call add_json_bool, Needs_text_relocations, $(filter true,$(TARGET_NEEDS_PLATF
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_str,  Target_exfat_driver, $(TARGET_EXFAT_DRIVER))
 $(call add_json_str_omitempty, Target_process_sdk_version_override, $(TARGET_PROCESS_SDK_VERSION_OVERRIDE))
+$(call add_json_str_omitempty, Target_specific_headers_include_dir, $(TARGET_SPECIFIC_HEADER_PATH))
 $(call add_json_str, Target_shim_libs, $(TARGET_LD_SHIM_LIBS))
 $(call add_json_bool, Uses_generic_camera_parameter_library, $(if $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY),,true))
 $(call add_json_bool, Uses_nvidia_enhancements, $(filter TRUE,$(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)))
-- 
2.17.1

